pkg/bwagreement/test/server_test.go

```diff
@@ -21,18 +22,20 @@ import (
 	"storj.io/storj/satellite/satellitedb/satellitedbtest"
 )
 
-func TestSameSerialNumberBandwidthAgreements(t *testing.T) {
+func TestBandwidthAgreement(t *testing.T) {
 	satellitedbtest.Run(t, func(t *testing.T, db satellite.DB) {
 		ctx := testcontext.New(t)
 		defer ctx.Cleanup()
 
-		/* More than one storage node can submit bwagreements with the same serial number.
-		   Uplink would like to download a file from 2 storage nodes.
-		   Uplink requests a PayerBandwidthAllocation from the satellite. One serial number for all storage nodes.
-		   Uplink signes 2 RenterBandwidthAllocation for both storage node. */
-		satellitePubKey, satellitePrivKey, uplinkPrivKey := generateKeys(ctx, t)
-		server := bwagreement.NewServer(db.BandwidthAgreement(), zap.NewNop(), satellitePubKey)
+		testDatabase(ctx, t, db.BandwidthAgreement())
+	})
+}
+
+func testDatabase(ctx context.Context, t *testing.T, bwdb bwagreement.DB) {
```

Comment:
I like all these tests. These are great. But why not put them in separate functions and have testDatabase() call them all? The way this is, it takes a lot of careful review to be pretty sure that each step won't affect local variables or the server state in an important way.

Reply:
Because of tparse, it's being quite noisy in Travis and don't want to add too much to it. However, yeah, once tparse shows information better, then it would be nice to have them separately.

Reply:
ok then